framework/src/userobject/ElementSubdomainModifier.C

```diff
@@ -368,3 +372,26 @@ ElementSubdomainModifier::setOldAndOlderSolutionsForMovedNodes(SystemBase & sys)
   if (older_solution)
     older_solution->close();
 }
+
+void
+ElementSubdomainModifier::setAncestorsSubdomainIDs(const SubdomainID & subdomain_id,
```

Comment:
This should be:

```
void
ElementSubdomainModifier::setAncestorsSubdomainIDs(Elem & elem) const
{
  Elem * current_elem = &elem;
  unsigned int lv = current_elem->level();

  for (unsigned int i = lv; i > 0; --i)
  {
    current_elem = curent_elem->parent();
    current_elem->subdomain_id() = elem.subdomain_id();

    // displaced parent element
    Elem * displaced_elem = _displaced_problem ? _displaced_problem->mesh().elemPtr(id) : nullptr;
    if (displaced_elem)
      displaced_elem->subdomain_id() = elem.subdomain_id();
  }
}
```

There's no need to ever call `_mesh.elemPtr()`. `current_elem->parent()` should return a non-const elem because `current_elem` is non const